collections/src/main.rs

```diff
@@ -41,17 +46,19 @@ fn list_med(list: &[i32]) -> ListResult {
                         max = Some((key, val))
                     }
                 }
-                None => max = Some((key, val))
+                None => max = Some((key, val)) // FIXME
             }
         }
         match max {
             Some((key, _)) => *key,
-            None => 0
+            None => 0 // FIXME
         }
     };
 
     ListResult { avg, median, mode }
 }
 
 #[derive(Debug)]
-struct ListResult { avg: i32, median: i32, mode: i32 }
\ No newline at end of file
+struct ListResult { avg: i32, median: i32, mode: i32 }
+
+// TODO: tests
```

Comment:
Cover your implementation with tests. 

"Code is cheap, prove its correctness" (c)

Reply:
In Rust Book, unit testing is in chapter 11, although the first tasks in chapter 8.3. :)
Added tests now. How it should be named? Linter doesn't seem to like ```subject__case__expected_result``` scheme.

Reply:
I'd rather go with simple BDD approach: just describe the tested behaviour. 

Example:
```rust
#[cfg(test)]
mod hash_spec {
    use super::*;
    
    #[test]
    fn has_64_symbols_len() {
        assert_eq!(hash("some_string").len(), 64);
    }
    
    #[test]
    fn contains_hex_chars_only() {
        assert!(!hash("some_string").contains("z"));
    }
}
```